### ISSUE TITLE
VGG32 not giving any predictions after training on same datasets as other models

### ISSUE THREAD

**Issue by SjoerdBraaksma** [divamgupta/image-segmentation-keras#317]
Hello!

I am using the various models of this repo for my thesis. I try to predict zebra crossings from aerial images. It has the following setup:

1) I trained 6 different models in 6 different datasets.
2) evaluated the models by letting them predict the same test set.

However, I notice that the VGG32 model gives normal predictions for 2 datasets, and for the other 4 it fails to predict anything at all. I just get blank outcomes!

This, while during training the epoch notifications show a high accuracy.

As an example, I ran the code below on the same image, using the weights from training on the small and large dataset. Along with the ground truth.

All other models provide correct predictions on the same images, and have been trained on the same images. Heck, even the VGG32 model with the small and small + transformed data performs fine, just the models with the larger dataset don't. It's really weird, and I am at a loss at what the possible reason could be. Any help is greatly appreciated!

```
# load weights small
weightfile = base_path+"data/train_checkpoints/VGG32/small/VGG32_small_weights.00005"

VGG32.load_weights(filepath = weightfile)

VGG32.predict_segmentation(
    inp= "/content/gdrive/MyDrive/data_science/MBA/Thesis/data/testset/images/21025.png",
    out_fname= "/content/gdrive/MyDrive/data_science/MBA/Thesis/data/testset/test_small.png"
    )

image_small = cv2.imread("/content/gdrive/MyDrive/data_science/MBA/Thesis/data/testset/test_small.png")
plt.imshow(image_small)



# load weights large
![example](https://user-images.githubusercontent.com/41111399/127053514-403f83cc-5a19-4824-b6d7-a22634fdf7f3.png)
![example_train](https://user-images.githubusercontent.com/41111399/127053635-e29aacd4-40c1-4825-95d0-ad9f4f981f74.png)

weightfile = base_path+"data/train_checkpoints/VGG32/large/VGG32_large_weights.00005"

VGG32.load_weights(filepath = weightfile)

VGG32.predict_segmentation(
    inp= "/content/gdrive/MyDrive/data_science/MBA/Thesis/data/testset/images/21025.png",
    out_fname= "/content/gdrive/MyDrive/data_science/MBA/Thesis/data/testset/test_large.png"
    )

image_large = cv2.imread("/content/gdrive/MyDrive/data_science/MBA/Thesis/data/testset/test_large.png")
plt.imshow(image_large)
```

**Comment by SjoerdBraaksma** [comment]
anyone has an idea? help would be kindly appreciated!

**Comment by ssnirgudkar** [comment]
I have the same problem with my vgg implementation

**Comment by varungupta31** [comment]
@ssnirgudkar @SjoerdBraaksma I'm having some issues with the vgg model as well, may I know the tensorflow version you are using to train the model?
